cmd/githubgen/codeowners.go

```diff
@@ -0,0 +1,195 @@
+# <component_path_relative_from_project_root>/<min_1_space><owner_1><space><owner_2><space>..<owner_n>
+# extension/oauth2clientauthextension/                 @open-telemetry/collector-contrib-approvers @pavankrish123 @jpkrohling
+# Path separator and minimum of 1 space between component path and owners is
+# important for validation steps
+#
+
+* @open-telemetry/collector-contrib-approvers
+`
+
+type codeownersGenerator struct {
+}
+
+func (cg codeownersGenerator) generate(data *githubData) error {
+	allowlistData, err := os.ReadFile(data.allowlistFilePath)
+	if err != nil {
+		return err
+	}
+	allowlistLines := strings.Split(string(allowlistData), "\n")
+
+	allowlist := make(map[string]struct{}, len(allowlistLines))
+	for _, line := range allowlistLines {
+		allowlist[line] = struct{}{}
+	}
+	var missingCodeowners []string
+	var duplicateCodeowners []string
+	members, err := getGithubMembers()
+	if err != nil {
+		return err
+	}
+	for _, codeowner := range data.codeowners {
+		_, present := members[codeowner]
+
+		if !present {
+			_, allowed := allowlist[codeowner]
+			allowed = allowed || strings.HasPrefix(codeowner, "open-telemetry/")
+			if !allowed {
+				missingCodeowners = append(missingCodeowners, codeowner)
+			}
+		} else if _, ok := allowlist[codeowner]; ok {
+			duplicateCodeowners = append(duplicateCodeowners, codeowner)
+		}
+	}
+	if len(missingCodeowners) > 0 {
+		sort.Strings(missingCodeowners)
+		return fmt.Errorf("codeowners are not members: %s", strings.Join(missingCodeowners, ", "))
+	}
+	if len(duplicateCodeowners) > 0 {
+		sort.Strings(duplicateCodeowners)
+		return fmt.Errorf("codeowners members duplicate in allowlist: %s", strings.Join(duplicateCodeowners, ", "))
+	}
+
+	codeowners := codeownersHeader
+	deprecatedList := "## DEPRECATED components\n"
+	unmaintainedList := "\n## UNMAINTAINED components\n"
+
+	unmaintainedCodeowners := unmaintainedHeader
+	currentFirstSegment := ""
+LOOP:
+	for _, key := range data.folders {
+		m := data.components[key]
+		for stability := range m.Status.Stability {
+			if stability == unmaintainedStatus {
+				unmaintainedList += key + "/\n"
+				unmaintainedCodeowners += fmt.Sprintf("%s/%s @open-telemetry/collector-contrib-approvers \n", key, strings.Repeat(" ", data.maxLength-len(key)-1))
+				continue LOOP
+			}
+			if stability == "deprecated" && (m.Status.Codeowners == nil || len(m.Status.Codeowners.Active) == 0) {
+				deprecatedList += key + "/\n"
+			}
+		}
+
+		if m.Status.Codeowners != nil {
+			parts := strings.Split(key, string(os.PathSeparator))
+			firstSegment := parts[0]
+			if firstSegment != currentFirstSegment {
+				currentFirstSegment = firstSegment
+				codeowners += "\n"
+			}
+			owners := ""
+			for _, owner := range m.Status.Codeowners.Active {
+				owners += " "
+				owners += "@" + owner
+			}
+			codeowners += fmt.Sprintf("%s/%s @open-telemetry/collector-contrib-approvers%s\n", key, strings.Repeat(" ", data.maxLength-len(key)-1), owners)
```

Comment:
I think this needs to still be `data.maxLength-len(key)`, as both `data.maxLength` and `len(key)` are now shorter after removal of the trailing slash from [main.go:224](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/28655/files#diff-95967fd59d5322fb780bc484d6d7f7a7b5bc5f4712e59a53304a60675eb6ae7cL224).

Same in line 126 above.
```suggestion
			codeowners += fmt.Sprintf("%s/%s @open-telemetry/collector-contrib-approvers%s\n", key, strings.Repeat(" ", data.maxLength-len(key)), owners)
```